sys/cloudformation/stack.yaml

```diff
@@ -165,370 +230,343 @@ Resources:
+      Name: !Sub '${ServiceName}-elb'
+      Subnets: !Ref Subnets
       SecurityGroups:
-      - !Ref 'sgelb'
+        - !Ref sgelb
       Tags:
         - Key: env
-          Value: badge-poser
+          Value: !Ref ServiceName
 
   # ELB LISTENER
   elblistener80:
     Type: AWS::ElasticLoadBalancingV2::Listener
-    DependsOn:
-    - elb
     Properties:
       DefaultActions:
         - Type: fixed-response
           FixedResponseConfig:
-            ContentType: "text/plain"
-            MessageBody: "aws is a teapot"
-            StatusCode: "418"
-      LoadBalancerArn: !Ref 'elb'
+            ContentType: 'text/plain'
+            MessageBody: 'AWS is a teapot'
+            StatusCode: '418'
+      LoadBalancerArn: !Ref elb
       Port: 80
       Protocol: HTTP
 
   elblistener443:
     Type: AWS::ElasticLoadBalancingV2::Listener
-    DependsOn:
-    - elb
     Properties:
       DefaultActions:
         - Type: fixed-response
           FixedResponseConfig:
-            ContentType: "text/plain"
-            MessageBody: "https teapot"
-            StatusCode: "418"
-      LoadBalancerArn: !Ref 'elb'
+            ContentType: 'text/plain'
+            MessageBody: 'https teapot'
+            StatusCode: '418'
+      LoadBalancerArn: !Ref elb
       Port: 443
       Protocol: HTTPS
       SslPolicy: ELBSecurityPolicy-2016-08
       Certificates:
-        - CertificateArn: !Ref 'ELBCertificateArn'
+        - CertificateArn: !Ref ELBCertificateArn
 
   # ECS CLUSTER
   ecscluster:
     Type: AWS::ECS::Cluster
     Properties:
-      ClusterName:
-        Fn::Sub:
-          - "${ServiceName}-cluster-${Environment}"
+      ClusterName: !Sub '${ServiceName}-cluster-${Environment}'
       CapacityProviders:
         - FARGATE
         - FARGATE_SPOT
       Tags:
         - Key: env
-          Value: badge-poser
+          Value: !Ref ServiceName
 
   # ECS SERVICE
   ecsservice:
-    Type: 'AWS::ECS::Service'
-    DependsOn:
-    - ecscluster
-    - ecstask
-    - elb
-    - elbtargetgroup
+    Type: AWS::ECS::Service
     Properties:
-      Cluster: !Ref 'ecscluster'
+      Cluster: !Ref ecscluster
       DesiredCount: 1
       HealthCheckGracePeriodSeconds: 15
       LaunchType: FARGATE
       LoadBalancers:
         - ContainerName: 'nginx'
           ContainerPort: 80
-          TargetGroupArn: !Ref 'elbtargetgroup'
-      ServiceName: !Ref 'ServiceName'
-      TaskDefinition: !Ref 'ecstask'
+          TargetGroupArn: !Ref elbtargetgroup
+      ServiceName: !Ref ServiceName
+      TaskDefinition: !Ref ecstask
       NetworkConfiguration:
         AwsvpcConfiguration:
           AssignPublicIp: ENABLED
-          Subnets: !Ref 'Subnets'
+          Subnets: !Ref Subnets
           SecurityGroups:
-          - !Ref 'sgecs'
+            - !Ref sgecs
       Tags:
         - Key: env
-          Value: badge-poser
+          Value: !Ref ServiceName
 
   # ECS TASK DEFINITION
   ecstask:
-    Type: 'AWS::ECS::TaskDefinition'
-    DependsOn:
```

Comment:
is this useless? just to know more about it ;)

Reply:
Yep, the dependence is already enforced by this:
```

          LogConfiguration:
            LogDriver: awslogs	            LogDriver: awslogs
            Options:	            Options:
              awslogs-group: !Ref 'cloudwatchloggroup'	              awslogs-group: !Ref cloudwatchloggroup
```